### PR TITLE
Update documentation for both README and crate

### LINKED ISSUE #4: Is this usable, what's the API?

### PR DESCRIPTION
**Pull request by edvbld**
Given that #7, #8 and #9 now has been merged, I figured it was a good time to update the documentation :smile: I tried to provide an example using each kind of API (convenience functions, `InflateWriter` and `InflateStream`). What do you guys think?

This also closes #4

### CODE REVIEW

**Comment by edvbld** [comment]
Thanks again for hooking up Travis! Feel very free to squash this PR if you decide to merge it :smile:

File: README.md [hunk 1]
```diff
@@ -50,7 +50,7 @@ while n < data.len() {
     let res = inflater.update(&data[n..]);
     if let Ok((num_bytes_read, result)) = res {
         n += num_bytes_read;
-        out.extend(result);
+        out.extend(result.iter().map(|v| *v));
```
Comment (eddyb) [inline README.md]: `.iter().cloned()` should also work, right? (same in the other location)

**Comment by nwin** [comment]
I’ll skim through it later. Can you please also un-pup the module `writer`? It doesn’t make much sense to have a 1-type module publicly exposed.

File: README.md (hunk 1 above)
Comment (nwin) [inline README.md]: Probably, couldn’t remember what was the most elegant way to solve it back then…

**Comment by edvbld** [comment]
@nwin sure, I created #13. Once/if #13 lands I can rebase this PR on top of `master`. There is absolutely no rush in reviewing, I'm in no hurry in getting these patches in. Thanks @eddyb, @oyvindln and @nwin for reviewing!

File: README.md (hunk 1 above)
Comment (edvbld) [inline README.md]: @eddyb yep, `.iter().cloned()` works as well. I pushed 5ca5c77 if you want to check it out

**Comment by edvbld** [comment]
Thanks @nwin for pushing #13, I just rebased this PR on top of `master`

**Comment by nwin** [comment]
Thank you. 🙂

**Pull request merged**